client/state/plugins/wporg/reducer.js

```diff
@@ -45,7 +71,58 @@ export function items( state = {}, action ) {
 	}
 }
 
+export function lists( state = {}, action ) {
+	const { category, data, searchTerm, type } = action;
+	switch ( type ) {
+		case PLUGINS_WPORG_LIST_RECEIVE:
+			if ( data ) {
+				// We only need lists by category and search terms.
+				if ( category ) {
+					const prevCategoryState = state.category?.[ category ] ?? [];
+					return {
+						...state,
+						category: {
+							...state.category,
+							[ category ]: [ ...prevCategoryState, ...data ],
```

Comment:
I may be wrong but would this be a good fit for a `keyedReducer`?

Reply:
It would be if we were using objects, but here we gather the plugins in arrays. Also, in the case where we're dealing with paginated data, we don't overwrite previous data, but we only append the new data.

I was thinking to build a couple of functions to remove repetition here, but eventually left it like that to make to more clear as to what we're doing and remove unnecessary indirection.

Do you have any further suggestions or do you think we can leave it as-is?